src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -196,16 +193,31 @@ internal CommitManager(ITextView textView, RecentItemsManager recentItemsManager
             {
                 // We are on the wrong thread.
                 FatalError.ReportWithoutCrash(new InvalidOperationException("Subject buffer did not provide Edit Access"));
-                return AsyncCompletionData.CommitBehavior.None;
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
             if (subjectBuffer.EditInProgress)
             {
                 FatalError.ReportWithoutCrash(new InvalidOperationException("Subject buffer is editing by someone else."));
-                return AsyncCompletionData.CommitBehavior.None;
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
+            }
+
+            CompletionChange change;
+
+            // We met an issue when external code threw an OperationCanceledException and the cancellationToken is not cancelled.
+            // Catching this scenario for further investigations.
+            try
+            {
+                change = completionService.GetChangeAsync(document, roslynItem, completionListSpan, commitCharacter, cancellationToken).WaitAndGetResult(cancellationToken);
+            }
+            catch (OperationCanceledException e) when (e.CancellationToken != cancellationToken)
+            {
+                FatalError.ReportWithoutCrash(e);
```

Comment:
You want this to go inside the when expression so you can get the report before the stack unwinds. ReportWithoutCrash intentionally returns Boolean so you can do things like:

`catch (OperationCancelledException e) when (e.CancellationToken != cancellationToken && ReportWithoutCrash(e))`